lib/chef/provider/package/apt.rb

```diff
@@ -255,6 +273,81 @@ def package_data_for(pkg)
           }
         end
 
+        # Helper to construct Hash of names-to-package-information.
```

Comment:
Everything from here down looks like its copied from `Package::Dpkg`. We don't want to duplicate that. We already have the shared `Package::Deb` mixin between the Dpkg and Apt providers that we'd want to refactor this into.

Reply:
I understand that most of the implementation is being referred from dpkg package. I will make use of `Package::Deb` and will separate out commonly used code by having a mixin between dpkg and apt providers.